packages/extender/src/actions/core/DisplayAction.spec.jsx

```diff
@@ -0,0 +1,309 @@
+import React, {useEffect, useState} from 'react';
+import {DisplayAction} from './DisplayAction';
+import {registry} from '../../registry';
+import {mount} from 'enzyme';
+import {ButtonRenderer} from '../samples/ButtonRenderer';
+import {LinkRenderer} from '../samples/LinkRenderer';
+import PropTypes from 'prop-types';
+import {act} from 'react-dom/test-utils';
+
+jest.useFakeTimers();
+
+describe('DisplayAction', () => {
+    beforeEach(() => {
+        registry.clear();
+    });
+
+    it('should call onClick when button is click', () => {
+        const action = registry.addOrReplace('action', 'test-action-1', {
+            label: 'Simple action',
+            onClick: jest.fn()
+        });
+
+        const wrapper = mount(<DisplayAction actionKey="test-action-1"
+                                             context={{path: '/test1'}}
+                                             render={ButtonRenderer}/>);
+
+        expect(action.onClick.mock.calls.length).toBe(0);
+        wrapper.find('button').simulate('click');
+        expect(action.onClick.mock.calls.length).toBe(1);
+    });
+
+    it('should call method with different contexts', () => {
```

Comment:
This test is too big and not so unitary, you can split it into 6 different tests.